packages/strapi/lib/core-api/__tests__/service.test.js

```diff
@@ -148,3 +167,23 @@ describe('Default Service', () => {
     });
   });
 });
+
+describe('getFetchParams', () => {
+  test.each([
+    [{ _limit: '1' }, 1],
+    [{ _limit: '0' }, 0],
+    [{ _limit: 0 }, 0],
+    [{ _limit: '' }, 0], // if _limit specified as ?_limit=, return 0
+    [{ _limit: '500' }, 50], // if _limit exceeds max allowed limit, return max allowed limit
+    [{ _limit: '-1' }, 50], // -1 should return all items but max limit is set, so return max allowed limit
+    [{}, 20], // if _limit not specified, return default limit
+    [{ _limit: 1000 }, 1000], // if max _limit is not specified, return requested _limit
```

Comment:
can you pass maxLimit here instead of doing a special case ? and just setting it if defined in the input :)

Reply:
yeah, I was thinking of that too, but I didn't like the idea much as it would require modification of the `getLimitParam` just for the test purposes 🤔 Also, I'm not sure how exactly `params` are passed from the request down here, but I'm afraid this way a user could do something like `/some-api-call?_limit=500&maxLimit=1000` and `maxLimit` would be accepted, or no? Correct me if I'm wrong.

Reply:
let me share want I meant :) Just make some logic in the test but don't change the code :)

```js
test.each([

])('Tes _limit %s', (input, expect) => {
  strapi.config.api.rest.maxLimit = input.maxLimit;
  expect(createService.getFetchParams({ _limit: input._lmit }).toMatch(...)

})
```

Reply:
I see 😅 alright, will update it :)

Reply:
Have a look now, I've also updated descriptions so it's more clear now :) 😇

Reply:
![image](https://user-images.githubusercontent.com/11503453/104006659-10e89b80-51a7-11eb-9024-f0b1d15aac36.png)